### ISSUE TITLE
[Bug🐛] Graceful broker and controller shutdown reports unhealthy observability after blocking scope admission

### ISSUE THREAD

**Issue by mxsm** [mxsm/rocketmq-rust#10643]
### Prerequisites

- [x] I have searched existing issues and discussions
- [x] This is a bug, not a question (use Discussions for questions)

### Environment

- OS: Windows 11 and Ubuntu 24.04, x86_64
- RocketMQ Rust: main (50e1a864a)
- Rust: 1.95.0

### Bug Description

`1f910acab` (bounded blocking scope admission and drain authorization) scopes every child `ChildServiceContext` blocking lane to its own task group, so a shut-down scope now refuses ordinary submissions with `runtime.context.unavailable`. Owner finalization that submits blocking I/O after its service task group has already closed is refused, and the affected shutdown component is reported unhealthy:

- **Broker**: `service_tasks` is shut down before the observability flush, so the telemetry flush submitted through `service_context.metadata_io()` fails. `broker_transactional_startup::listener_bind_failure_rolls_back_already_started_components` fails with `rollback should be healthy: ["observability"]` on Linux CI, and the Windows-only `broker_process_startup` test observes exit code 70 from a real Broker process after `/drainz`.
- **Controller**: `finish_controller_process_shutdown` shuts `service_context.task_group()` down before the telemetry flush, so draining a locally started Controller exits 70 with `controller.internal.failure`.
- **NameServer**: the runtime shuts the shared service task group down once the route manager has routed work through it, and the flush that follows `boot_with_lifecycle` is then refused. A fresh instance that never initialized that group still exits 0, so this branch is conditional.

Two Broker unit tests also fail in the same CI run:

- `broker::log_filter_control::tests::broker_log_filter_control_child` expects `BrokerLogFilterControlError::Scheduling` for a stopped TTL controller, but `apply()` writes the intent audit before probing the controller, so the coincident audit failure is reported instead of the typed scheduling rejection.
- `broker_runtime::tests::scheduled_tasks_shutdown_precedes_exclusive_message_store_shutdown` still searches `rocketmq-broker/src/broker_runtime/lifecycle.rs` for `let scheduled_report = self`, renamed to `scheduled_outcome` when the scheduled shutdown report was unified.

### Steps to Reproduce

```sh
cargo test -p rocketmq-broker --lib
cargo test -p rocketmq-broker --no-fail-fast
```

The owner finalization failure can also be observed on a real process: start a Broker, Controller, or NameServer with `ROCKETMQ_HEALTH_BIND_ADDR` set, request `GET /drainz`, and inspect the exit code (70 before the fix).

### Expected vs Actual

**Expected:** The final telemetry flush submits under an allowance reserved while the owning scope was still open, the observability component stays healthy, and a drained Broker or Controller exits 0.
**Actual:** The flush is refused with `runtime.context.unavailable`, the observability component is reported unhealthy, and the process exits 70.

### Logs/Screenshots

```text
"logs_shutdown_error": "owned blocking provider shutdown failed: runtime.context.unavailable: Runtime context is unavailable"
Broker lifecycle shutdown report is unhealthy unhealthy_components=["observability"]
ERROR core.service.failed: Service lifecycle operation failed
```

```text
assertion failed: matches!(error, BrokerLogFilterControlError::Scheduling(_))
broker_runtime::tests::scheduled_tasks_shutdown_precedes_exclusive_message_store_shutdown:
  scheduled task shutdown should exist
```

### Additional Context

`BlockingDrainLease` documents itself as the bounded authorization for already accepted work and owner finalization, but had no production caller. The fix reserves one submission while each scope is still open and flushes through it, so the scope keeps refusing new work while its owner still finishes: a shared `reserve_telemetry_flush_lease` helper plus `TelemetryRuntimeGuard::shutdown_with_drain_lease`, wired into the Broker lifecycle, the Controller bootstrap, and the NameServer bootstrap. MCP already flushes before closing its scope and stays unchanged.
